tests_e2e/tests/scripts/agent_cgroups_process_check-unknown_process_check.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env pypy3
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script forces the process check by putting unknown process in the agent's cgroup
+
+import os
+import subprocess
+import datetime
+
+from assertpy import fail
+
+from azurelinuxagent.common.utils import shellutil
+from tests_e2e.tests.lib.cgroup_helpers import get_agent_cgroup_mount_path, \
+    BASE_CGROUP, check_agent_quota_disabled, check_log_message
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.retry import retry_if_false
+
+
+def prepare_agent():
+    check_time = datetime.datetime.utcnow()
+    log.info("Executing script update-waagent-conf to enable agent cgroups config flag")
+    result = shellutil.run_command(["update-waagent-conf", "Debug.CgroupCheckPeriod=20", "Debug.CgroupLogMetrics=y",
+                                    "Debug.CgroupDisableOnProcessCheckFailure=y",
+                                    "Debug.CgroupDisableOnQuotaCheckFailure=n"])
+    log.info("Successfully enabled agent cgroups config flag: {0}".format(result))
+
+    found: bool = retry_if_false(lambda: check_log_message(" Agent cgroups enabled: True", after_timestamp=check_time))
+    if not found:
+        fail("Agent cgroups not enabled")
+
+
+def creating_dummy_process():
+    log.info("Creating dummy process to add to agent's cgroup")
+    dd_command = ["dd", "if=/dev/zero", "of=/dev/null"]
+    proc = subprocess.Popen(dd_command)
+    return proc.pid
+
+
+def remove_dummy_process(pid):
+    log.info("Removing dummy process from agent's cgroup")
+    shellutil.run_command(["kill", "-9", str(pid)])
+
+
+def disable_agent_cgroups_with_unknown_process(pid):
+    """
+    Adding dummy process to the agent's cgroup and verifying that the agent detects the unknown process and disables cgroups
+
+    Note: System may kick the added process out of the cgroups, keeps adding until agent detect that process
+    """
+    def unknown_process_found():
+        cgroup_procs_path = os.path.join(BASE_CGROUP, "cpu,cpuacct" + get_agent_cgroup_mount_path(), "cgroup.procs")
+        log.info("Adding dummy process %s to cgroup.procs file %s", pid, cgroup_procs_path)
+        process = subprocess.Popen(f"echo {pid} > {cgroup_procs_path}", shell=True)
```

Comment:
need to check for errors writing to the file. you can use run_command or just python's open/write functions